packages/backend/src/auto-peering/service.ts

```diff
@@ -56,8 +68,107 @@ export async function createAutoPeeringService(
   }
 
   return {
-    acceptPeeringRequest: (args: PeeringRequestArgs) =>
-      acceptPeeringRequest(deps, args)
+    acceptPeeringRequest: (args) => acceptPeeringRequest(deps, args),
+    initiatePeeringRequest: (args) => initiatePeeringRequest(deps, args)
+  }
+}
+
+async function initiatePeeringRequest(
+  deps: ServiceDependencies,
+  args: InitiatePeeringRequestArgs
+): Promise<Peer | AutoPeeringError> {
+  const { peerUrl } = args
+
+  const asset = await deps.assetService.get(args.assetId)
+
+  if (!asset) {
+    return AutoPeeringError.UnknownAsset
+  }
+
+  const outgoingHttpToken = uuid()
+
+  const peeringDetailsOrError = await sendPeeringRequest(deps, peerUrl, {
+    ilpConnectorAddress: deps.config.ilpConnectorAddress,
+    staticIlpAddress: deps.config.ilpAddress,
+    asset: { code: asset.code, scale: asset.scale },
+    httpToken: outgoingHttpToken,
```

Comment:
Why not let the Rafiki instance receiving the auto-peering request generate that token?

Reply:
Oh, ok. Maybe I didn't read the code correctly.

Reply:
Talked in person, each peer will present its own outgoing token